angular/projects/global/src/lib/utils/utils.ts

```diff
@@ -22,4 +30,90 @@ export class Utils {
     // trim the last |
     return values.replace(/\|$/, '');
   }
+
+  public formatDate(date) {
+    if (date) {
+      const d = new Date(date);
+      let month = '' + (d.getMonth() + 1);
+      let day = '' + d.getDate();
+      const year = d.getFullYear();
+      if (month.length < 2) {
+        month = '0' + month;
+      }
+      if (day.length < 2) {
+        day = '0' + day;
+      }
+      return [year, month, day].join('');
+    } else {
+      return null;
+    }
+  }
+
+  public convertJSDateToNGBDate(jSDate: Date) {
+    return {
+      year: jSDate.getFullYear(),
+      month: jSDate.getMonth() + 1,
+      day: jSDate.getDate()
+    };
+  }
+
+  public convertFormGroupNGBDateToJSDate(nGBDate, nGBTime = null) {
+    if (nGBTime === null) {
+      return new Date(nGBDate.year, nGBDate.month - 1, nGBDate.day);
+    } else {
+      return new Date(nGBDate.year, nGBDate.month - 1, nGBDate.day, nGBTime.hour, nGBTime.minute);
+    }
+  }
+
+  public formatBytes(bytes, decimals = 2) {
+    if (bytes == null) {
+      return '-';
+    }
+    if (bytes === 0) {
+      return '0 Bytes';
+    }
+
+    const k = 1024;
+    const dm = decimals < 0 ? 0 : decimals;
+    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+
+    const i = Math.floor(Math.log(bytes) / Math.log(k));
+
+    return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
+  }
+
+  public encodeParams(inputParam: string) {
+    return encode(inputParam)
+      .replace(/\(/g, '%28')
+      .replace(/\)/g, '%29');
+  }
+
+  public encodeFilename(filename: string, isUrl: boolean) {
+    if (isUrl) {
+      return encode(filename)
+        .replace(/\(/g, '%28')
+        .replace(/\)/g, '%29')
+        .replace(/\\/g, '_')
+        .replace(/\//g, '_')
+        .replace(/\%2F/g, '_')
+        .replace(/ /g, '_');
+    } else {
+      return filename
+        .replace(/\(/g, '%28')
+        .replace(/\)/g, '%29')
+        .replace(/\\/g, '_')
+        .replace(/\//g, '_');
+    }
+  }
```

Comment:
Is there any reason why we wouldn't just use something like: 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

to handle encoding?

Reply:
Toss this as it's not being used.